### ISSUE TITLE
Idea : Waypoint system (TS like) for RA

### ISSUE THREAD

**Issue by ShadowA** [OpenRA/OpenRA#10470]
I saw in the TS sneak peek video (https://www.youtube.com/watch?v=yrt_m-xwbok) the icon of the "waypoint" mode, will be implemented in the TS mod, won't it ?

I was wondering if this would be transferrable to RA. 

I know you can do this with Shift+clicks to draw a route. 
However, drawing a "waypoints route" once and for all would be really awesome, as it would help : 

- having them visually, and be sure I didn't forget Shift while clicking (messes up the whole thing)
- setting predefined routes (to avoid something for instance, or to sneak)
- using again the same route at different times, if old army destroyed
- and most importantly ^_^ setting infinite patrol loops

**Comment by abcdefg30** [comment]
Duplicate of #2598, I think.

**Comment by ShadowA** [comment]
Hmm not exactly, if I'm not mistaken those are temporary waypoints (RA2 style), 

I was thinking more about permanent waypoint route (TS like) 

The route would never disappear, unless you explicitly delete it (or delete some of the flags)
And you could use it again, just by right-clicking at the first flag of the route (or any other flag)

I'm not sure if it's the same as #2598 , maybe I misunderstood it then.

**Comment by pchote** [comment]
This is indeed different to #2598.

**Comment by ShadowA** [comment]
@pchote All good then, thanks ! :)

**Comment by Punsho** [comment]
Waypoints are already implemented #16549. They aren't persistent though

**Comment by matjaeck** [comment]
This could get fixed by #17021.

**Comment by obrakmann** [comment]
[This](https://github.com/OpenRA/OpenRA/compare/bleed...obrakmann:patrol) might be relevant as well, at least for the last bullet point.

**Comment by pchote** [comment]
Following on from a Discord discussion about @obrakmann's patrol branch, I hacked together a proof of concept as to how we could upgrade a closed loop of attack moves into a patrol activity: 4993cd69cc24accbc06e92580032e07a4250dcc3

**Comment by penev92** [comment]
This issue is asking for a TS waypoints for several reasons, most of which we seem to be covering in different, arguably better ways with rally point paths for production structures, order queuing for units and (hopefully incoming) patrol mode.
The only thing we don't cover is "preset paths that can be reused by units", but is that really worth the hassle of implementing?

**Comment by dragunoff** [comment]
> The only thing we don't cover is "preset paths that can be reused by units", but is that really worth the hassle of implementing?

I don't think this is worth implementing because it doesn't fit the fast gameplay of CNC games (no wonder it was dropped after TS). The situation on the map changes very frequently so having a re-usable waypoint is not worth much. One-time waypoints (aka Shift-queuing) and patrol are way more useful.

**Comment by penev92** [comment]
Then once we have patrolling we can consider this issue resolved IMO.
